packages/eslint-plugin/src/rules/no-deprecated-components.ts

```diff
@@ -0,0 +1,172 @@
+/* !
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ */
+
+import { TSESTree } from "@typescript-eslint/utils";
+
+import { createRule } from "./utils/createRule";
+
+const DEPRECATED_TO_NEW_MAPPING: { [deprecated: string]: string } = {
+    AbstractComponent: "AbstractComponent2",
+    AbstractPureComponent: "AbstractPureComponent2",
+    CollapsibleList: "OverflowList",
+    MultiSelect: "MultiSelect2",
+    PanelStack: "PanelStack2",
+    Popover: "Popover2",
+    Select: "Select2",
+    Suggest: "Suggest2",
+    Tooltip: "Tooltip2",
+};
+const PACKAGES_WITH_DEPRECATED_IMPORTS = ["@blueprintjs/core", "@blueprintjs/select"];
+
+type MessageIds = "nonDeprecated" | "deprecated";
+
+// tslint:disable object-literal-sort-keys
+export const noDeprecatedComponentsRule = createRule<unknown[], MessageIds>({
```

Comment:
I think there should be some mention here (even if just for devs inspecting the rule source) of the fact that we're not actually reading the `@deprecated` JSDoc annotation, and thus the rule is quite distinct from the `deprecation/deprecation` ESLint rule.